_posts/en/newsletters/2023-04-12-newsletter.md

```diff
@@ -0,0 +1,181 @@
+---
+title: 'Bitcoin Optech Newsletter #246'
+permalink: /en/newsletters/2023/04/12/
+name: 2023-04-12-newsletter
+slug: 2023-04-12-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME
+
+## News
+
+- **Splicing specification discussions:** several LN developers posted
+  to the Lightning-Dev mailing list this week about ongoing work on the
+  [draft specification][bolts #863] for [splicing][topic splicing],
+  which allows some funds in an offchain LN channel to be spent onchain
+  (splice-out) or for onchain funds to be added to an offchain channel
+  (splice-in).  The channel can continue to operate while an onchain
+  splicing transaction is waiting for sufficient confirmations.
+
+  {:.center}
+  ![Splicing transaction flow](/img/posts/2023-04-splicing1.dot.png)
+
+  Discussions this week included:
+
+  - *Which commitment signatures to send:* when a splice is created, a
+    node will hold parallel commitment transactions, one that spends
+    from the original funding output and one that spends from each new
+    funding output for all pending splices.  As of this writing, the
+    current draft specification involves sending signatures for every
+    commitment transaction that could possibly need to be confirmed,
+    even for commitment transactions whose signatures have previously
```

Comment:
> It sounds like the exact same data is sent again

That is correct.

> but wouldn’t you just need to develop two parallel sets of commitment transactions (one based on the old channel anchor, the other based on the new channel anchor with the splice) until the splice has sufficient confirmation?

That is also correct.

> Perhaps it could be clarified why any data might be duplicated.

"As Lisa Neigut [explains][neigut splice], this allows reusing the workflow already used in LN for updating a commitment transaction."

The current (pre-splice) workflow for updating a commitment transaction is for Alice to send Bob her signature for the new commitment (and vice versa Bob) and then Alice to send Bob her revocation point for the old commitment (also vice versa Bob).  For splices, there's multiple alternative commitment transactions that all need to be kept in sync, so instead of sending individual signatures and revocation points, Alice and Bob each send an array of sigs and revocations for all of the current alternatives.

When you create a new alternative commitment, you only **need** to send signatures for it.  But the pre-splice workflow and the original splice design expect you to use a message that sends signatures for every current alternative commitment transaction (and revocations for every previous state).  That's unnecessarily redundant, but changing the protocol to only send two required signatures involves some tradeoffs that apparently developers wanted to consider, specifically that adding a different message, or using the existing messages in different contexts, increases the complexity of the protocol and requires writing more tests.

(I'll try to work at least some of the above into the text, but I wanted to explain it here to see if it makes sense to you or to see if any LN people show up to tell me I'm daft.  :-)

Reply:
Yeah, it wasn’t clear to me whether the other parallel state was being sent in addition, or the old states, so your longer explanation makes this much clearer to me :)

Thanks for explaining!

Reply:
Quick note that what I described to @Xekyo above is not entirely accurate.  The current splicing draft doesn't send an array of signatures or revocation points; instead, those messages are duplicated for every current commitment transaction.  The general point holds, though: the simplest protocol/testing situation would be "when sending sigs for any commitment transaction, send one for all current commitment transactions".  To only send a single sig for a new splice requires special casing in several places in the specifacation (and presumably in implementations).

I just wanted to note that here in case anyone will be confused by my update to the newsletter text differing from what I described here.  Glad I stopped to actually read the spec before writing about things.  :-D